genai_processors/contrib/openrouter_model.py

```diff
@@ -0,0 +1,424 @@
+
+  max_tokens: int | None
+  """Maximum number of tokens to generate."""
+
+  response_format: dict[str, Any] | None
+  """Response format specification (e.g., {"type": "json_object"})"""
+
+  stop: list[str] | str | None
+  """Stop sequences to end generation."""
+
+  stream: bool
+  """Whether to stream the response. Defaults to True."""
+
+  tools: list[dict[str, Any]] | None
+  """Function calling tools available to the model."""
+
+  tool_choice: str | dict[str, Any] | None
+  """How the model should use tools ("auto", "none", specific tool)"""
+
+  logit_bias: dict[str, float] | None
+  """Modify likelihood of specific tokens."""
+
+  transforms: list[str] | None
+  """OpenRouter-specific transforms to apply."""
+
+  models: list[str] | None
+  """Fallback models if primary model fails."""
+
+  route: Literal['fallback'] | None
+  """Routing strategy for model selection."""
+
+  provider: dict[str, Any] | None
+  """Provider-specific preferences."""
+
+
+def _to_openrouter_message(
+    part: content_api.ProcessorPart, default_role: str = 'user'
+) -> dict[str, Any]:
+  """Convert ProcessorPart to OpenRouter message format."""
+  role = part.role.lower() if part.role else default_role
+
+  # Handle function calls
+  if part.function_call:
+    return {
+        'role': part.role.lower(),
+        'function_call': {
+            'name': part.function_call.name,
+            'arguments': json.dumps(part.function_call.args),
+        },
+    }
+
+  # Handle function responses
+  if part.function_response:
+    return {
+        'role': 'function',
+        'name': part.function_response.name,
+        'content': json.dumps(part.function_response.response),
+    }
+
+  # Handle text content
+  if content_api.is_text(part.mimetype):
+    return {
+        'role': role,
+        'content': part.text,
+    }
+
+  # Handle multimodal content
+  if content_api.is_image(part.mimetype):
+    # Convert image to base64 for OpenRouter
+    import base64
+
+    if part.part.inline_data:
+      encoded_image = base64.b64encode(part.part.inline_data.data).decode(
+          'utf-8'
+      )
+      return {
+          'role': role,
+          'content': [
+              {
+                  'type': 'image_url',
+                  'image_url': {
+                      'url': f'data:{part.mimetype};base64,{encoded_image}'
+                  },
+              }
+          ],
+      }
+
+  # Fallback to text representation
+  return {
+      'role': role,
+      'content': str(part.part),
+  }
+
+
+def _parse_sse_line(line: str) -> dict[str, Any] | None:
+  """Parse a Server-Sent Events line."""
+  line = line.strip()
+
+  # Skip comments and empty lines
+  if not line or line.startswith(':'):
+    return None
+
+  # Parse data lines
+  if line.startswith('data: '):
+    data = line[6:]
+    if data == '[DONE]':
+      return {'type': 'done'}
+
+    try:
+      return json.loads(data)
+    except json.JSONDecodeError:
```

Comment:
Is there a reason to suppress errors? I'd expect that receiving a malformed JSON would mean that something went wrong and we can't be sure that server would behave robustly in the future. So may be just don't handle the error and let the pipeline crash.

That is if errors are not part of regular flow in openrouter, I don't have much experience with it.